Create ponyc options

winksaville opened the pull request:
Refactor the option processing code in main to libponyc/options. The hope
is that this will allow easier testing of standard ponyc options in the
future.

jemc commented:
Can you check my review from the last time you posted this?
https://github.com/ponylang/ponyc/pull/2432#discussion_r157906140

SeanTAllen commented:
I dont understand from the comments how this makes testing easier. Can you squash this to a single commit with an explanatory "how this makes testing easier" commit comment @winksaville ? Given there are no tests as part of this commit, its not at all obvious to me that this makes things easier to test.

jemc commented:
Also, please check my review comment linked to above.

winksaville commented:
I've squashed this and updated the commit message.

@jemc, I believe I've already addressed the concerning the common prefix. If I've missed something please add a comment directly to this change, txs.

winksaville commented:
I've been asked to add a compiler flag for openssl, if this PR is acceptable can we merge it as I'd like to add the the new flag on top of this change. Otherwise I'll close this so its not hanging around.

jemc commented:
CI failures seem unrelated, so I'm merging.
